cdc/scheduler/internal/scheduler.go

```diff
@@ -52,7 +52,22 @@ type Scheduler interface {
 	// It is thread-safe
 	Rebalance()
 
-	// Close scheduler and release it's resource.
+	// DrainCapture is used to drop all tables situated at the target capture
+	// It is thread-safe
+	DrainCapture(target model.CaptureID) int
+
+	// Close scheduler and release resource.
 	// It is not thread-safe.
 	Close(ctx context.Context)
 }
+
+// Query is for scheduler related owner job.
+// at the moment, only for `DrainCapture`, we can use this to handle all manual schedule task.
+type Query struct {
+	CaptureID model.CaptureID
+
+	// changefeedID model.ChangeFeedID
+	// tableID      model.TableID
```

Comment:
Dead code?

Reply:
`ChangefeedID` and `TableID` is useful if we refact `MoveTable` API to use `Query`.